src/api/vdom.js

```diff
@@ -258,7 +267,7 @@ const newElementOpenEnd = wrapIdomFunc(elementOpenEnd);
 
 // Standard open / closed overrides don't need to reproduce internal behaviour
 // because they are the ones referenced from *End and *Start.
-const newElementOpen = wrapIdomFunc(elementOpen, stackOpen);
+const newElementOpen = attributesContext(wrapIdomFunc(elementOpen, stackOpen));
```

Comment:
The idea would be to pass `attributes` in to every patch and patch would look to that as the source of truth instead of the global.